### PR TITLE
Require integration test to be run as sudo/root

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
As discussed in https://github.com/k3s-io/k3s/pull/4372, this rips out all the "sudo" commands within the K3s integration utility functions. It is now required that integration tests have root permissions when run.
Updated documentation on this.

#### Types of Changes ####
Test Fix

#### Verification ####
As sudo user:
`sudo -E env "PATH=$PATH" go test ./pkg/... ./tests/integration/... -run Integration` 
As root:
` go test ./pkg/... ./tests/integration/... -run Integration`

#### Linked Issues ####
N/A internal testing change only, nothing for QA to do.

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/util/cmd.go [hunk 1]
```diff
@@ -66,32 +67,17 @@ func IsExistingServer() bool {
 //   cmdEx1, err := K3sCmd("etcd-snapshot", "ls")
 //   cmdEx2, err := K3sCmd("kubectl", "get", "pods", "-A")
 func K3sCmd(cmdName string, cmdArgs ...string) (string, error) {
+	if !IsRoot() {
+		return "", fmt.Errorf("integration tests must be run as sudo/root")
```
Comment (briandowns) [inline tests/util/cmd.go:71]:
```suggestion
		return "", errors.New("integration tests must be run as sudo/root")
```

File: tests/util/cmd.go [hunk 2]
```diff
@@ -140,6 +126,10 @@ func FindStringInCmdAsync(scanner *bufio.Scanner, target string) bool {
 // with the provided arguments. Subsequent/parallel calls to this function will block until
 // the original lock is cleared using K3sKillServer
 func K3sStartServer(inputArgs ...string) (*K3sServer, error) {
+	if !IsRoot() {
+		return nil, fmt.Errorf("integration tests must be run as sudo/root")
```
Comment (briandowns) [inline tests/util/cmd.go:130]:
```suggestion
		return nil, errors.New("integration tests must be run as sudo/root")
```

**Review by brandond — APPROVED** [review]
LGTM once Brian's nits are addressed.

File: tests/util/cmd.go (hunk 1 above)
Comment (dereknola) [reply tests/util/cmd.go]: Fixed!

File: tests/util/cmd.go (hunk 2 above)
Comment (dereknola) [reply tests/util/cmd.go]: Fixed!

**Review by briandowns — APPROVED** [review]

**Pull request merged**